libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/location/LocationViewModel.kt

```diff
@@ -22,24 +24,24 @@ sealed class LocationAction {
 class LocationViewModel : UIViewModel<Location?, LocationAction>(null) {
     val navigationLocationProvider = NavigationLocationProvider()
 
+    /**
+     * Accessor function to get the last [Point]
+     */
     val lastPoint: Point?
         get() = navigationLocationProvider.lastLocation?.run {
             Point.fromLngLat(longitude, latitude)
         }
 
-    private val locationObserver: LocationObserver = object : LocationObserver {
-        override fun onNewRawLocation(rawLocation: Location) {
-            // We are not sending LocationAction.UpdateRaw here, because we are expecting to
-            // receive locationMatcherResult
-        }
-
-        override fun onNewLocationMatcherResult(locationMatcherResult: LocationMatcherResult) {
-            navigationLocationProvider.changePosition(
-                location = locationMatcherResult.enhancedLocation,
-                keyPoints = locationMatcherResult.keyPoints,
-            )
-            invoke(LocationAction.Update(locationMatcherResult.enhancedLocation))
+    /**
+     * Suspend until a non-null location is available.
+     */
+    suspend fun firstLocation(): Location {
+        var nonNullLocation: Location? = navigationLocationProvider.lastLocation
+        while (nonNullLocation == null) {
+            delay(DELAY_FIRST_LOCATION_MS)
```

Comment:
What is the reason we are adding a delay?
@kmadsen

Reply:
Okay looks like we are waiting unless we get a non null location. It makes the experience nice in a sense that the location puck rendering is very fast vs if we don't wait for non null location, then we start at the ocean and then transition to current location.

Reply:
```
Delays coroutine for a given time without blocking a thread and resumes it after a specified time.
```

This allows the coroutine to wait without blocking the main thread. Maybe we'll find a better way to handle this when handling location permissions.